### PR TITLE
fixing #184

### LINKED ISSUE #184: audit 2.3.2 no longer supports `chef-server` fetcher + `chef-server-visibility` collector
### Cookbook version
2.3.2

### Chef-client version
any

### Platform Details
any

### Scenario:

```
$ cat attributes/default.rb
default['audit']['collector'] = 'chef-server-visibility'
default['audit']['fetcher'] = 'chef-server'
default['audit']['insecure'] = true
default['audit']['quiet'] = false

default['audit']['profiles'] = [{name: "rhel-6-L1",compliance: "cis/cis-rhel6-level1"}]
```

### Steps to Reproduce:
see above

### Expected Result:
Profiles should be fetched from Compliance - they are not since the `/compliance` prefix is not added.

### Actual Result:
```
[2016-12-16T14:39:50+00:00] DEBUG: Initiating GET to https://z-chef.lxc/organizations/demo/owners/cis/compliance/cis-rhel6-level1/tar
.
.
.
[2016-12-16T14:39:50+00:00] ERROR: 404 "Not Found" (Net::HTTPServerException)
```

In audit 2.2.0, the above GET is:
```
[2016-12-16T14:33:23+00:00] DEBUG: Initiating GET to https://z-chef.lxc/compliance/organizations/demo/owners/cis/compliance/cis-rhel6-level1/tar
```

### PR DESCRIPTION
**Pull request by jeremymv2**
### Description

This allows `chef-server-visibility` collector + `chef-server` fetcher to work as desired.

### Issues Resolved

#184 

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

Signed-off-by: Jeremy J. Miller <jm@chef.io>

### CODE REVIEW

**Comment by jeremymv2** [comment]
Still need to add unit tests for `files/default/vendor/chef-server/fetcher.rb` still to prevent regression.

**Comment by jeremymv2** [comment]
unit tests added

**Comment by jeremymv2** [comment]
@alexpop Are you able to help with these travis-ci build errors perhaps?

```
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Failed to complete #create action: [unable to sign request without credentials set] on default-amzn
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

**Comment by alexpop** [comment]
Looks like external PRs are not getting the encrypted ENV variables needed for the kitchen-ec2 testing:
https://github.com/travis-ci/travis-ci/issues/1938

Will need to give you access to the repo @jeremymv2 so that you can create branches and submit PRs like that.

In the meantime, I'm testing this locally on my machine.

File: files/default/vendor/chef-server/fetcher.rb [hunk 1]
```diff
@@ -50,11 +50,8 @@ def self.chef_server_org
     end
 
     def self.url_prefix
-      if chef_server_visibility?
-        ''
-      else
-        '/compliance'
-      end
+      return '/compliance' unless chef_server_visibility? && Chef.node.attributes['audit']['fetcher'] != 'chef-server'
```
Comment (alexpop) [inline files/default/vendor/chef-server/fetcher.rb]: This method is getting harder to understand. 
I pinged Victoria as well and would like to keep it as if-then-else:

```ruby
def self.url_prefix
  if chef_server_visibility? && Chef.node.attributes['audit']['fetcher'] != 'chef-server'
    ''
  else
    '/compliance'
  end
end
```

**Comment by alexpop** [comment]
I tested it and the kitchen-ec2 tests pass. Can you please squash the commits?

Thanks a lot for the contribution!

**Comment by chris-rock** [comment]
Thank you @jeremymv2 and @alexpop

**Pull request merged**